lib/modules/datasource/pypi/index.ts

```diff
@@ -246,7 +246,7 @@ export class PypiDatasource extends Datasource {
     const releases: Releases = {};
     for (const link of Array.from(links)) {
       const version = PypiDatasource.extractVersionFromLinkText(
-        link.text,
+        link.text.trim(),
```

Comment:
```suggestion
        link.text?.trim(),
```

Need to defensively make sure it's a string first or it will throw an error. Perhaps better done earlier/more verbosely with an is.string() if it's not already checked somewhere prior